### PR TITLE
Fix tool for creating review for GitHub PR by zanuda's report

### PR DESCRIPTION
**Pull request by s-khechnev**
### Related
[Branch with tool](https://github.com/Kakadu/zanuda/tree/review)

## Changes
- Implemented creating review by zanuda's rdjsonl report
  - Implemented correct numbering of lines in diff
  - Added handling of extra cases in diff lines, such as `@@ -1 +1 @@`, `\ No newline at end of file`, `--- a/main.ml`
  - Refactored `review/main.ml` by adding options in `info` type
  - Introduced inline tests for diff lines parser
  - Introduced cram tests for diff parser
  - Removed dependency on `Opal` in `review/line_parser.ml`. Now using only `Angstrom`
- Added reviewer's package in dune-project.

### Example of usage
- [workflow file](https://github.com/s-khechnev/demo-review/blob/a1cf4475610cb949e9ec6cb36f37131ff48d4481/.github/workflows/PrPost.yml)
- [demo review](https://github.com/s-khechnev/demo-review/pull/1)

### CODE REVIEW

File: review/dune [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ (inline_tests)
+ (preprocess
+  (pps ppx_inline_test ppx_show ppx_expect)))
+
+(executable
+ (name parser)
+ (modules parser)
+ (libraries diff_parser)
+ (preprocess
+  (pps ppx_show)))
```
Comment (Kakadu) [inline review/dune:27]: И тут тоже нужно?

File: review/main.ml [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    Format.printf "headers: %a\n" Curly.Header.pp x.Curly.Response.headers;
+    Format.printf "body: %s\n" x.Curly.Response.body
+  | Error e -> Format.printf "Failed: %a" Curly.Error.pp e
+;;
+
+(* let your_token =
+  match Sys.getenv "TOKEN" with
+  | s -> s
+  | exception Not_found -> failwith "Setup token using env TOKEN"
+;; *)
```
Comment (Kakadu) [inline review/main.ml:16]: Удалить?

File: review_tests/test2.t/run.t [hunk 3]
```diff
@@ -0,0 +1,2 @@
+  $ cat test.diff | ../../review/parser.exe - -f test.js -l 21
+  Got something. It should be at 25 line in diff
```
Comment (Kakadu) [inline review_tests/test2.t/run.t:2]: А почему это правильный вывод?

File: review/dune [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ (name main)
+ (public_name reviewer)
+ (modules main)
+ (libraries diff_parser curly))
+
+(library
+ (name diff_parser)
+ (modules types line_parser diff_parser)
+ (wrapped false)
+ (libraries angstrom stdio yojson ppx_show.runtime)
```
[the remark this answers is not in the document]
Comment (s-khechnev) [reply review/dune:17]: https://github.com/Kakadu/zanuda/pull/22/files#diff-c213a2ca273aab686e6f21e7ad7d6ad35e2e89172cb21c4ab9ebfff7e768495dR98

**Pull request merged**